### ISSUE TITLE
Time to train is considerably large

### ISSUE THREAD

**Issue by Desuka-art** [innnky/so-vits-svc#120]
So, I have a 3090, and yet when I train, it takes about 4 hours to get to 4000 steps. I was wondering what I should do to fix this and make it train at the speed it's supposed to train on.

**Comment by G-Sea** [comment]
The speed of training is related to the number of datasets, the more datasets, the slower the training speed. You can modify the "batch_size" in the config.json , change "6" to a larger number.

**Comment by Desuka-art** [comment]
What would be a good batch size for a 3090, in your opinion?

**Comment by G-Sea** [comment]
> 

14。
If something goes wrong, try lower number, like 12

**Comment by Desuka-art** [comment]
Thank you! Anything else I should do to make it the fastest it can go?

**Comment by Likkkez** [comment]
> What would be a good batch size for a 3090, in your opinion?

Why did the guy say 14, I set it to 23 on my 3090ti and theres still memory to spare. You can probably even get away with 24 if u don't have anything else on it.

**Comment by Likkkez** [comment]
just check for memory usage in nvtop and u can figure out the perfect number. But i'm pretty sure its about 1 batch size = 1 gigabyte and 3090 has 24 gigabytes

**Comment by Desuka-art** [comment]
Thank you! Does it also depend on what Python version I'm using? Because I'm using 3.8.

**Comment by Likkkez** [comment]
> Thank you! Does it also depend on what Python version I'm using? Because I'm using 3.8.

I don't think so, that would be kinda weird

**Comment by Desuka-art** [comment]
I'm back. I tried that, it's the exact same speed. I thought it might have to do with num_workers, but when I go above 4, it gives me an OOM error. I don't know what to do. Any suggestions?

**Comment by TomoGaSukunai** [comment]
Try to slice your dataset into smaller pieces, and trim the voiceless segments.  It seems the longer segment desires larger memories.

> I'm back. I tried that, it's the exact same speed. I thought it might have to do with num_workers, but when I go above 4, it gives me an OOM error. I don't know what to do. Any suggestions?

**Comment by Desuka-art** [comment]
But I don't have that many files, and the biggest file is only 10 seconds.

**Comment by Likkkez** [comment]
> But I don't have that many files, and the biggest file is only 10 seconds.

Do you have the files on HDD? I remember when i had it on HDD it was very very slow and the GPU was waiting for the files to load most of the time. You should move it to SSD if thats the case

**Comment by Desuka-art** [comment]
No, the files are on an SSD. I don't understand what's going on...

**Comment by Desuka-art** [comment]
I think I figured it out... it's using both my CPU AND GPU together in tandem, what do I do to fix it?
